cogito/protocol.go

```diff
@@ -48,6 +49,39 @@ type PutRequest struct {
 	Env    Environment
 }
 
+func (req *PutRequest) UnmarshalJSON(data []byte) error {
+	type request PutRequest // Alias to avoid infinite loops.
```

Comment:
See also
- https://www.orsolabs.com/post/go-json-default-values/
- https://eli.thegreenplace.net/2020/optional-json-fields-in-go/

Reply:
To wrap your mind around this, start with Source.UnmarshalJSON() at line 126 below.

Reply:
It is not obvious for sure! Your explanation is correct.

You see? One day you will even start appreciating Go :trollface:

Reply:
Just to clarify additionally for me one more thing:
It does not find it because the function signature are different?
`func (req *PutRequest) UnmarshalJSON(data []byte) error`
vs
`func (request PutRequest) UnmarshalJSON(data []byte) error`
?

Reply:
Almost. The other one would not be `func (request PutRequest) UnmarshalJSON(data []byte) error` but `func (req *request) UnmarshalJSON(data []byte) error`